src/java.base/share/classes/java/net/NetMulticastSocket.java

```diff
@@ -216,9 +216,8 @@ public synchronized void bind(SocketAddress addr) throws SocketException {
             throw new SocketException("already bound");
         if (addr == null)
             addr = new InetSocketAddress(0);
-        if (!(addr instanceof InetSocketAddress))
```

Comment:
in the context of this type of change the negative logic is a little obtuse or disjoint in the new style variable declaration, and its subsequent use. I'd find it a more natural flow and easier to read with the instance pattern prefacing a block in which the variable is used
if (addr instanced InetSocketAddress epoint) {
  // a block which is using the epoint variable
   if (epoint.isUnresolved)
      throw new SocketException("Unresolved address");
   InetAddress iaddr = epoint.getAddress;
 etc, etc ...
} else {
   throw new IllegalArgumentException("Unsupported address type!");
}

Reply:
The disadvantage is that it would add another level of braces.

Reply:
maybe ... I find it would provide an easier flow to code  ... an "easy on the eye" call flow linkage - the variable is declared and then used  .... instead of the disconnect of throwing an exception or non immediate use of the variable. 

but sure,  isn't code somewhat idiosyncratic!!

Reply:
The first few times I looked at this pattern, I did stumble over it somewhat. But since then, I find it natural enough and easy to follow. Maybe the concern is transient in nature, until the instanceof pattern matching feature is more widely used.